examples/binned_spikes/main.cpp

```diff
@@ -0,0 +1,177 @@
+#include <memory>
+#include <fstream>
+#include <iomanip>
+
+#include "science/scipp/status.h"
+#include "science/synapse/channel.h"
+#include "science/synapse/data.h"
+#include "science/synapse/device.h"
+#include "science/synapse/nodes/electrical_broadband.h"
+#include "science/synapse/nodes/stream_out.h"
+#include "science/synapse/nodes/spike_detect.h"
+#include "science/synapse/nodes/spectral_filter.h"
+
+using synapse::BinnedSpiketrainData;
+using synapse::Ch;
+using synapse::ElectricalBroadbandData;
+using synapse::SynapseData;
+
+auto stream(const std::string& uri, const std::string& group, bool configure) -> int {
+  synapse::Device device(uri);
+  synapse::Config config;
+  science::Status s;
+
+  synapse::DeviceInfo info;
+  s = device.info(&info);
+  if (!s.ok()) {
+    std::cout << "error getting device info: (" << static_cast<int>(s.code()) << ") " << s.message() << std::endl;
+    return 1;
+  }
+  std::cout << "device info: " << info.DebugString() << std::endl;
+
+  std::shared_ptr<synapse::StreamOut> stream_out{};
+  if (configure) {
+    stream_out = std::make_shared<synapse::StreamOut>("out", group);
+    std::vector<synapse::Ch> channels;
+    for (unsigned int i = 1; i < 97; i++) {
+      channels.push_back(synapse::Ch{
+        .id = i,
+        .electrode_id = i * 2,
+        .reference_id = i * 2 + 1,
+      });
+    }
+    auto electrical_broadband = std::make_shared<synapse::ElectricalBroadband>(
+        1, channels, 16, 30000, 20.0, 500, 6000);
+    s = config.add_node(electrical_broadband);
+
+    auto spectral_filter = std::make_shared<synapse::SpectralFilter>(
+        synapse::SpectralFilterMethod::kBandPass, 500, 6000);
+
+    s = config.add_node(spectral_filter);
+    s = config.connect(electrical_broadband, spectral_filter);
+
+    auto channel_mask = synapse::ChannelMask(0); // TODO(kevinc): should this be number of channels?
```

Comment:
this should take a list of channel ids to consume / run spike detection on
seems like it's not used on the server side on the cart? which is fine; but if we merge this we should use it correctly here